beluga-tunnel/src/services.rs

```diff
@@ -0,0 +1,65 @@
+use core::net::SocketAddrV4;
+
+use tokio::io::{AsyncReadExt, AsyncWriteExt};
+use tokio::net::TcpStream;
+use tokio::sync::mpsc::{Receiver, Sender};
+use tokio::task::JoinHandle;
+
+use crate::error::Error;
+use crate::Result;
+
+pub(crate) struct SshService {
```

Comment:
Can we turn this into a more generic "TCP pipe"? Apart from the hard-coded socket address nothing in here is specific to ssh.

Reply:
Yeah, let's create this interface(trait) that will handle such stuff.